File: backend/src/models/Expense.js
import mongoose from 'mongoose';

const expenseSchema   = new mongoose.Schema(
    {
        title: {type: String, required: true, trim: true},
        amount: { type: Number, required: true },
        // receipts: [{ type: String, required: true }],
        category: { type: String, required: true },
        payingMember: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
        groupId: { type: mongoose.Schema.Types.ObjectId, ref: 'Group', required: true },
        membersInvolveds: [{
            member: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
            ponderation: { type: Number, default: 1 }
        }],
        date: {type: Date, default: Date.now}        
    },
    { timestamps: true }
)

export default mongoose.model('Expense', expenseSchema)

File: Frontend/mobiles/app/src/main/java/com/example/commoncash/domain/model/Expense.kt
package com.example.commoncash.domain.model

import java.util.Date

data class Expense(
    var title: String,
    var amount: Int,
    // var receipts:String,
    var category: String,
    var payingMember: User,
    var groupId: Group? = null,
    var membersInvolveds: List<Pair<User, Int>>? = null,
    var date: Date

)
